File: 6-Tarea-Arreglos-matrices/B.Alicia-y-el-cachorro-saltarin.cpp
#include <iostream>
using namespace std;

int main() {
  int N;
  cin >> N;

  int saltos[N];
  for (int i = 0; i < N; i++) {
    cin >> saltos[i];
  }

  int A;
  cin >> A;

  int cachorro = 0, respuesta = 0;
  for (int i = 0; i < N; i++) {
    cachorro += saltos[i];

    if (cachorro == A)
      respuesta++;
  }

  cout << respuesta << endl;

  return 0;
}

File: 6-Tarea-Arreglos-matrices/B.Alicia-y-el-cachorro-saltarin.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);

    int N = scanner.nextInt();
    int[] saltos = new int[N];

    for (int i = 0; i < N; i++)
      saltos[i] = scanner.nextInt();

    int A = scanner.nextInt();

    int cachorro = 0, respuesta = 0;
    for (int i = 0; i < N; i++) {
      cachorro += saltos[i];

      if (cachorro == A)
        respuesta++;
    }

    System.out.println(respuesta);
  }
}